base_layer/core/tests/helpers/sync.rs

```diff
@@ -140,53 +155,77 @@ pub async fn create_network_with_local_and_peer_nodes() -> (
 #[allow(dead_code)]
 #[derive(Debug)]
 pub enum WhatToDelete {
+    BlocksAndHeaders,
     Blocks,
     Headers,
 }
 
+// Private helper function to setup a delete a block transaction.
+// Note: This private function will panic if the index is out of bounds - caller function's responsibility.
+fn delete_block(txn: &mut DbTransaction, node: &NodeInterfaces, blocks: &[ChainBlock], index: usize) {
+    txn.delete_block(*blocks[index].hash());
+    txn.delete_orphan(*blocks[index].hash());
+    txn.set_best_block(
+        blocks[index + 1].height(),
+        blocks[index + 1].accumulated_data().hash,
+        blocks[index + 1].accumulated_data().total_accumulated_difficulty,
+        *node.blockchain_db.get_chain_metadata().unwrap().best_block(),
+        blocks[index + 1].to_chain_header().timestamp(),
+    );
+}
+
 // Delete blocks and headers in reverse order; the first block in the slice wil not be deleted
 pub fn delete_some_blocks_and_headers(
     blocks_with_anchor: &[ChainBlock],
     instruction: WhatToDelete,
     node: &NodeInterfaces,
-    set_best_block: Option<bool>,
 ) {
     if blocks_with_anchor.is_empty() || blocks_with_anchor.len() < 2 {
         panic!("blocks must have at least 2 elements");
     }
-    let set_best_block = set_best_block.unwrap_or(false);
     let mut blocks: Vec<_> = blocks_with_anchor.to_vec();
     blocks.reverse();
     for i in 0..blocks.len() - 1 {
```

Comment:
```suggestion
    for i in (0..blocks.len() - 1).rev() {
```

Reply:
nit: if just do this, you don't have to swap around the entire array, and the code above is more direct what's going on. Because it seems strange that you delete a block, and then you set the next block as the best block